From 8964d6b3c98a96ac0bf11983d4513201a2a2b5eb Mon Sep 17 00:00:00 2001
From: Jerome Duval <jerome.duval@gmail.com>
Date: Mon, 19 Oct 2015 23:01:16 +0000
Subject: add Haiku as supported target.

* also replace haproxy-systemd-wrapper with $(EXTRA) in install-bin.

diff --git a/Makefile b/Makefile
index d15be04..d42d5fb 100644
--- a/Makefile
+++ b/Makefile
@@ -98,7 +98,7 @@ DOCDIR = $(PREFIX)/doc/haproxy
 # Use TARGET=<target_name> to optimize for a specifc target OS among the
 # following list (use the default "generic" if uncertain) :
 #    generic, linux22, linux24, linux24e, linux26, solaris,
-#    freebsd, openbsd, netbsd, cygwin, custom, aix51, aix52
+#    freebsd, openbsd, netbsd, cygwin, haiku, custom, aix51, aix52
 TARGET =
 
 #### TARGET CPU
@@ -220,6 +220,12 @@ ifeq ($(TARGET),generic)
   USE_POLL   = implicit
   USE_TPROXY = implicit
 else
+ifeq ($(TARGET),haiku)
+  # For Haiku
+  TARGET_LDFLAGS = -lnetwork
+  USE_POLL = implicit
+  USE_TPROXY = implicit
+else
 ifeq ($(TARGET),linux22)
   # This is for Linux 2.2
   USE_GETSOCKNAME = implicit
@@ -347,6 +353,7 @@ endif # linux26
 endif # linux24e
 endif # linux24
 endif # linux22
+endif # haiku
 endif # generic
 
 
diff --git a/README b/README
index 6d341d3..5bc5f04 100644
--- a/README
+++ b/README
@@ -48,6 +48,7 @@ and assign it to the TARGET variable :
   - aix51       for AIX 5.1
   - aix52       for AIX 5.2
   - cygwin      for Cygwin
+  - haiku       for Haiku
   - generic     for any other OS or version.
   - custom      to manually adjust every setting
 
-- 
2.2.2

